backend/x86_proc.ml

```diff
@@ -425,6 +381,11 @@ let reset_asm_code () =
   Section_name.Tbl.clear asm_code_by_section
 
 let generate_code asm =
+  (* Apply peephole optimizations to all sections *)
+  if !Oxcaml_flags.x86_peephole_optimize
+  then
+    X86_peephole_optimize.optimize_all_sections asm_code asm_code_by_section
```

Comment:
Is it necessary to optimize both `asm_code` and `asm_code_by_section`? It looks like they contain the same instructions and could be sectioned after optimization. Maybe that would require too much refactoring.